### ISSUE TITLE
0.11.0 skybox renders way too close to the camera

### ISSUE THREAD

**Issue by VirxEC** [JonahPlusPlus/bevy_atmosphere#75]
Updated from 0.10 to 0.11. My game works on a different scale than normal bevy games so I need to skybox to render far away from the camera, but in 0.11 it looks like this is broken as the skybox renders super close to the camera.

With the skybox:
![image](https://github.com/user-attachments/assets/3d549a80-e08a-4029-822b-741f01043bd0)

Without:
![image](https://github.com/user-attachments/assets/e747e9ec-5d45-4e18-8cc0-e06946bad9ba)

**Comment by ptsd** [comment]
Odd. The skybox mesh is generated with a size based on the far plane of the camera frustrum:
https://github.com/JonahPlusPlus/bevy_atmosphere/blob/da8c1e321ba3755ba78605b494385cfd16c73f41/src/plugin.rs#L114
https://github.com/JonahPlusPlus/bevy_atmosphere/blob/da8c1e321ba3755ba78605b494385cfd16c73f41/src/skybox.rs#L74

If i manually add:

```rust
Projection::from(
    PerspectiveProjection {
        far: 0.2,
        ..default()
})
```
to the camera entity in an example without a skybox I get strange behaviour:

![Screenshot 2024-12-10 at 12 15 44](https://github.com/user-attachments/assets/94ba8f1d-ce68-44f6-97b9-7f758a3a584f)

I would expect the cubes to be culled in this case as they are ~5.0 units from the camera, so this is possibly a bug in bevy itself? The same applies to things in your world beyond the default `far` value of `1000.0` so if you add a `Projection` component to your camera with an appropriate `far` value (like below) for your world this will presumably "fix" the issue?

```rust
Projection::from(
    PerspectiveProjection {
        far: 6000.0, // Or whatever is sensible for your scale...
        ..default()
})
```
Will raise this upstream with `bevy` 👍

**Comment by ptsd** [comment]
So it seems `PerspectiveProjection` not far culling is currently expected behaviour.

There was a somewhat forgotten PR (https://github.com/bevyengine/bevy/pull/8205) that I have updated (https://github.com/bevyengine/bevy/pull/16789).

This will make `CameraProjection::far()` an `Option<f32>` instead of a raw `f32` and default it to `None` (essentialy an infinite far plane) for `PerspectiveProjection`.

Assuming this goes in we probably want to think about allowing an override for the value of `far` passed to `skybox::mesh()` in the `atmosphere_insert` system?

**Comment by Shatur** [comment]
Closed by #76.
